webui/react/src/hooks/useSettings.ts

```diff
@@ -322,6 +320,7 @@ const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
 
   useLayoutEffect(() => {
     if (initialLoading) return;
+    clearQuerySettings();
```

Comment:
agree with @gt2345, this hook is used a lot and is prone to re-rendering. Can you double check to see if it's rendering minimally after these changes? 🙏 (It would be good to know also if they are not related to these changes but still re-rendering a bunch)

Reply:
✅ made it into a callback in latest commit